### DISCUSSION TITLE
System Resources on Baremetal

### DISCUSSION THREAD

**Discussion by lukeclements** [nasa/fprime#2311] [Q&A] [answered]
Hi, 

On a linux f' deployment I can access system resource information like RAM usage, CPU utilization, and more. I'm curious if a similar capability exists for baremetal?

**Comment by LeStarch** [comment, marked as answer]
I am working on a prototype version of this.  It does the following:

1. At start-up RAM is written with a pattern
2. Idle function detects start and end of pattern (between stack and heap). This is ~= free RAM.
3.  CPU usage ~= time spent in the idle function

I'll let you know when I get it in a state that it could be pulled into your project.

**Reply by lukeclements** [reply]
That sounds like what I am looking for. Will it be specific to certain devices since the hardware varies between different controllers?

**Reply by LeStarch** [reply]
Some features may be specific....but most of the above is fairly generic!
